### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by ddade** [iperov/DeepFaceLab#5464]
Apologies if this is not the proper mechanism.

I'm an amateur filmmaker, and I would like to experiment with replacing a live actor's face with a 3D model from software like Unreal Metahuman, or Daz 3D.

Since I can render photorealistic images, theoretically, I should be able to create very effective source training sets because I could match the destination lighting, facial expressions, and orientation of the head very closely.

For this reason, during the face extraction phase of the destination images, is it possible for the software to output a JSON file describing its best guess at the orientation of the head as a quaternion, Euler angles, or anything really, for each image? With that information, I could easily create an animation to provide the perfect source training data.

Is this possible? I'd like to discuss with a developer about adding this feature. I've got budget for it.

Aside from protecting production against actors who don't show up on shoot day, or later abandon their commitments as soon as they get a better gig, this would also solve the problem of retargeting facial expressions to the 3D model, a technical problem many would prefer to forget about.

Is there someone here with whom I could further discuss?

Thanks!

Thanks!

**Comment by iperov** [comment]
you don't need that. Just extract source faces.

**Comment by ddade** [comment]
Thank you for responding.

I think I see what you mean, but with that information about the orientation, I could greatly reduce the number of images I would have to render, couldn't I, by only rendering the orientations that are actually in the destination images? Given how long it takes to render a single photorealistic image, it would be an important optimization.

Or are you saying that that isn't how training works, and I really would have to generate a full range in any case? Given how long it takes to render a photorealistic frame, that might not be feasible.

This would be a huge deal for indie filmmakers.

Thanks again for responding.

**Comment by ddade** [comment]
Thanks again for responding.

No, I don't have to artistic talent to sculpt manually :) Here is a sample exported from Daz Studio and rendered in Blender.

I think what I can do is to do two takes, the second with a HMC to capture the actor's facial expression, and track the actor's head with another HTC Vive tracker puck. I won't have the headache of solving to a skeleton, just get the position an orientation, just like I already do to track the camera. I'll also be able to make the Blender scene's lighting match the real world set's lighting.

I think with that, I'll be able to create an animation that mimics reasonably closely the actual actor, with the correct orientation, facial expression, and lighting. I think this could represent a bare minimum. Then I think I'll use a Monte-Carlo-like method to create a fuller range, and let it keep adding varied images for as long as I can wait.

Off the cuff, do you think this is worth trying?
![sample_for_iperov](https://user-images.githubusercontent.com/1152172/150626231-3058b5fb-3406-41ee-857d-293b26627f90.png)

**Comment by iperov** [comment]
ok try. But eyes should be more realistic

**Comment by ddade** [comment]
Yes, there are better eye models, I turned down bounces so it would render faster, and that white parallelogram is a badly placed area light.

Thanks, I'll give it a try.

**Comment by iperov** [comment]
by the way Unreal Metahuman is the bomb. 
Can you sculpt for example some dead but popular historical person, and then render it with the same quality as in Metahuman, and generate a lot of facial expressions?

**Comment by ddade** [comment]
Yes, Metahumans are extremely impressive. But the process for getting the Metahuman facial rig, the thing that makes them so expressive, onto an arbitrary sculpt is still kind of being worked out. I've seen some people succeed spectacularly on Youtube, but it is not yet for the faint of heart. There is not yet any application that just automatically makes a sculpt into a Metahuman. I'm sure it's coming, though.

https://www.youtube.com/watch?v=tB0DOYbu8Og

Unreal Engine will be useful because it can render a Metahuman at that quality in pretty much realtime  and it'd be more feasible to generate thousands of images.
